master/internal/user/service.go

```diff
@@ -29,6 +30,53 @@ var forbiddenError = echo.NewHTTPError(
 	http.StatusForbidden,
 	"user not authorized")
 
+const (
+	// authNone indicates a request needs no authentication.
+	authNone int = 0
+	// authStandard indicates a request needs authentication.
+	authStandard = 1
+	// authAdmin indicates a request needs admin authentication.
+	authAdmin = 2
+)
+
+// unauthenticatedPointsList contains URIs that are exempted from authentication.
+var unauthenticatedPointsList = []string{
+	"/",
+	"/det/",
+	"/info",
+	"/task-logs",
+	"/ws/data-layer/.*",
+	"/agents",
+	"/det/.*",
+	"/login",
+	"/api/v1/master",
+	"/api/v1/auth/login",
+	"/api/v1/auth/logout",
+	"/proxy/:service/.*",
+	"/api/v1/notebooks/.*",
+	"/api/v1/tasks/.*",
+	"/api/v1/allocations/.*",
+	"/api/v1/tensorboards/.*",
+	"/api/v1/shells/.*",
+	"/api/v1/experiments/.*",
+	"/api/v1/trials/.*",
+	"/api/v1/checkpoints",
```

Comment:
drive by comment, why are any of these unauthed?

Reply:
When I was doing testing on this, things wouldn't run unless I had these unauthenticated - i.e. experiments wouldn't run unless `/api/v1/experiments/.*` was unauthenticated, or tensorboards wouldn't run without the tensorboard api path was unauthenticated. I think when the agent is running tasks, it hits some endpoints to send or receive info.

Reply:
shouldn't all new style APIs `/api/*` be authed through their own logic?

Reply:
however if this auth just checking if user is signed in then it's not clear why it'd be breaking.

Reply:
yes to the latter

Reply:
it could mean the harness is making unauthenticated calls, for example? For the purpose of security I'd say we should get to the bottom of this.

Reply:
ok after more testing, I can restrict the `/api/v1/experiments` path to `/api/v1/experiments/.*/model_def` and toss any paths unrelated to experiments (shells, notebooks, tasks). But I can't do much about the `trials`, `allocations`, or `checkpoint`.

Reply:
this sounds like all the calls the harness makes to the master?

Reply:
why can't you remove the model def call? that call (every call basically) should be authed.

Reply:
can you debug why that is happening? that endpoint should be authorized.

Reply:
Yeah looking into it - the harness needs the model_def, but the authorization doesn't see any token so responds with a 401 error.

Reply:
yeah, there should be a session token in the env it uses